build/common.sh

```diff
@@ -600,6 +600,7 @@ function kube::build::run_build_command_ex() {
     --env "GOFLAGS=${GOFLAGS:-}"
     --env "GOLDFLAGS=${GOLDFLAGS:-}"
     --env "GOGCFLAGS=${GOGCFLAGS:-}"
+    --env "SOURCE_DATE_EPOCH=${SOURCE_DATE_EPOCH:-}"
```

Comment:
I don't think we need this - all of the version info is saved outside of docker before we enter the build container because we don't sync `.git` into the container.

https://github.com/kubernetes/kubernetes/blob/170dcc2ea086f94fb5d80c75f2aa1c1039084d36/build/common.sh#L193-L194

Reply:
... though I guess it doesn't hurt, in case any other tools consult `SOURCE_DATE_EPOCH`.

Reply:
@ixdy the `kube::version::ldflag()` does not use the stuff from .dockerized-kube-version-defs

it builds a new date if the env var is not set (the best i can tell):
https://github.com/kubernetes/kubernetes/blob/master/hack/lib/version.sh#L155-L156

Reply:
The build image sets `KUBE_GIT_VERSION_FILE`:
https://github.com/kubernetes/kubernetes/blob/170dcc2ea086f94fb5d80c75f2aa1c1039084d36/build/build-image/Dockerfile#L36-L37

When this is set, we read it for the version info:
https://github.com/kubernetes/kubernetes/blob/170dcc2ea086f94fb5d80c75f2aa1c1039084d36/hack/lib/version.sh#L32-L38

Reply:
Oh wait, now I see what you're saying. Yes, you are right. I forgot that we only save the git metadata, not time metadata (unlike bazel's workspace status script).

Reply:
correct :)